File: app/src/main/java/com/spidroid/starry/activities/StoryContentFragment.java
package com.spidroid.starry.activities;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.media3.common.MediaItem;
import com.bumptech.glide.Glide;
// تم تحديث سطر الاستيراد هذا
import androidx.media3.exoplayer.ExoPlayer; // <--- التعديل هنا
//import com.google.android.exoplayer2.MediaItem;
import androidx.media3.ui.PlayerView;
import com.spidroid.starry.R;

// Fragment لعرض قصة واحدة (صورة أو فيديو)
public class StoryContentFragment extends Fragment { // تم جعل الكلاس public وفي ملفه الخاص
    private static final String ARG_MEDIA_URL = "media_url";
    private static final String ARG_MEDIA_TYPE = "media_type";
    private static final String ARG_CAPTION = "caption";

    private String mediaUrl;
    private String mediaType;
    private String caption;

    private ImageView ivStoryImage;
    private PlayerView pvStoryVideo;
    private ExoPlayer player;
    private TextView tvCaption;

    public static StoryContentFragment newInstance(String mediaUrl, String mediaType, String caption) {
        StoryContentFragment fragment = new StoryContentFragment();
        Bundle args = new Bundle();
        args.putString(ARG_MEDIA_URL, mediaUrl);
        args.putString(ARG_MEDIA_TYPE, mediaType);
        args.putString(ARG_CAPTION, caption);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mediaUrl = getArguments().getString(ARG_MEDIA_URL);
            mediaType = getArguments().getString(ARG_MEDIA_TYPE);
            caption = getArguments().getString(ARG_CAPTION);
        }
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        // يمكنك استخدام تخطيط مختلف هنا إذا كنت تريد تخطيطًا مختلفًا للصور والفيديوهات
        // أو دمج الاثنين في تخطيط واحد مع التحكم في الرؤية.
        // للتوضيح، سأفترض أن media_item_video يحتوي على PhotoView أو ImageView ليعرض الصور.
        View view = inflater.inflate(R.layout.media_item_video, container, false); // استخدام نفس التخطيط لـ PlayerView

        ivStoryImage = view.findViewById(R.id.photoView); // يجب أن يكون موجودًا في media_item_video أو layout آخر
        pvStoryVideo = view.findViewById(R.id.playerView);
        // ستحتاج لإضافة TextView للـ caption في الـ layout المستخدم (مثلاً media_item_video) أو ملف جديد
        tvCaption = view.findViewById(R.id.tv_story_caption_overlay); // تأكد من إضافة هذا الـ ID في media_item_video.xml

        // إخفاء/إظهار بناءً على نوع الميديا
        if ("image".equals(mediaType)) {
            if (pvStoryVideo != null) pvStoryVideo.setVisibility(View.GONE);
            if (ivStoryImage != null) {
                ivStoryImage.setVisibility(View.VISIBLE);
                Glide.with(this).load(mediaUrl).into(ivStoryImage);
            }
        } else if ("video".equals(mediaType)) {
            if (ivStoryImage != null) ivStoryImage.setVisibility(View.GONE);
            if (pvStoryVideo != null) {
                pvStoryVideo.setVisibility(View.VISIBLE);
                setupVideoPlayer();
            }
        }

        if (tvCaption != null) {
            if (caption != null && !caption.isEmpty()) {
                tvCaption.setText(caption);
                tvCaption.setVisibility(View.VISIBLE);
            } else {
                tvCaption.setVisibility(View.GONE);
            }
        } else {
            // يمكن إضافة Toast أو Log إذا لم يتم العثور على tvCaption
        }


        return view;
    }

    private void setupVideoPlayer() {
        if (player == null && getContext() != null) {
            player = new ExoPlayer.Builder(requireContext()).build();
            pvStoryVideo.setPlayer(player);
            player.setMediaItem(MediaItem.fromUri(mediaUrl));
            player.prepare();
            player.setPlayWhenReady(true);
            player.setRepeatMode(ExoPlayer.REPEAT_MODE_ONE); // تكرار الفيديو
        }
    }

    @Override
    public void onResume() {
        super.onResume();
        if (player != null) {
            player.setPlayWhenReady(true);
            // يمكنك أيضاً استئناف التقدم الزمني للقصة الرئيسية هنا إذا لزم الأمر
        }
    }

    @Override
    public void onPause() {
        super.onPause();
        if (player != null) {
            player.setPlayWhenReady(false);
        }
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        if (player != null) {
            player.release();
            player = null;
        }
    }
}

File: app/src/main/java/com/spidroid/starry/activities/StoryContentFragment.kt
// hmze123/sambok/sambok-main/app/src/main/java/com/spidroid/starry/activities/StoryContentFragment.kt
package com.spidroid.starry.activities

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ProgressBar
import androidx.core.os.bundleOf
import androidx.fragment.app.Fragment
import androidx.media3.common.MediaItem
import androidx.media3.common.PlaybackException // ✨ تم التأكد من الاستيراد
import androidx.media3.common.Player // ✨ تم التأكد من الاستيراد
import androidx.media3.exoplayer.ExoPlayer
import com.bumptech.glide.Glide
import com.spidroid.starry.R // ✨ تم التأكد من الاستيراد
import com.spidroid.starry.databinding.MediaItemVideoBinding
import com.spidroid.starry.models.StoryModel
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class StoryContentFragment : Fragment() {

    private var _binding: MediaItemVideoBinding? = null
    private val binding get() = _binding!!

    private var player: ExoPlayer? = null
    private var mediaUrl: String? = null
    private var mediaType: String? = null
    private var playWhenReady = true
    private var currentItem = 0
    private var playbackPosition = 0L

    private var loadingProgressBar: ProgressBar? = null

    // ✨ تم دمج كلا كتلي companion object هنا
    companion object {
        private const val TAG = "StoryContentFragment"
        private const val ARG_MEDIA_URL = "media_url"
        private const val ARG_MEDIA_TYPE = "media_type"

        @JvmStatic
        fun newInstance(mediaUrl: String, mediaType: String) =
            StoryContentFragment().apply {
                arguments = bundleOf(
                    ARG_MEDIA_URL to mediaUrl,
                    ARG_MEDIA_TYPE to mediaType
                )
            }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        arguments?.let {
            mediaUrl = it.getString(ARG_MEDIA_URL)
            mediaType = it.getString(ARG_MEDIA_TYPE)
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = MediaItemVideoBinding.inflate(inflater, container, false)
        loadingProgressBar = binding.root.findViewById(R.id.loadingProgressBar)
        return binding.root
    }

    private fun initializePlayer() {
        val safeContext = context ?: return
        val url = mediaUrl ?: return

        if (mediaType == StoryModel.MEDIA_TYPE_VIDEO) {
            player = ExoPlayer.Builder(safeContext)
                .build()
                .also { exoPlayer ->
                    binding.playerView.player = exoPlayer
                    val mediaItem = MediaItem.fromUri(url)
                    exoPlayer.setMediaItem(mediaItem)
                    exoPlayer.playWhenReady = playWhenReady
                    exoPlayer.seekTo(currentItem, playbackPosition)
                    exoPlayer.prepare()

                    exoPlayer.addListener(object : Player.Listener {
                        override fun onPlaybackStateChanged(state: Int) {
                            loadingProgressBar?.visibility = if (state == Player.STATE_BUFFERING) View.VISIBLE else View.GONE
                        }
                        // ✨ تم إضافة دالة onPlayerError المفقودة لتتوافق مع Player.Listener
                        override fun onPlayerError(error: PlaybackException) {
                            Log.e(TAG, "ExoPlayer error: ${error.message}", error)
                            // يمكنك إضافة Toast أو رسالة للمستخدم هنا
                        }
                    })
                }
            binding.playerView.visibility = View.VISIBLE
            binding.photoView.visibility = View.GONE
        } else {
            binding.playerView.visibility = View.GONE
            binding.photoView.visibility = View.VISIBLE
            Glide.with(this).load(url).into(binding.photoView)
        }
    }

    private fun releasePlayer() {
        player?.let { exoPlayer ->
            playbackPosition = exoPlayer.currentPosition
            currentItem = exoPlayer.currentMediaItemIndex
            playWhenReady = exoPlayer.playWhenReady
            exoPlayer.stop()
            exoPlayer.release()
        }
        player = null
    }

    override fun onStart() {
        super.onStart()
        if (androidx.media3.common.util.Util.SDK_INT > 23) {
            initializePlayer()
        }
    }

    override fun onResume() {
        super.onResume()
        if ((androidx.media3.common.util.Util.SDK_INT <= 23 || player == null)) {
            initializePlayer()
        }
    }

    override fun onPause() {
        super.onPause()
        if (androidx.media3.common.util.Util.SDK_INT <= 23) {
            releasePlayer()
        }
    }

    override fun onStop() {
        super.onStop()
        if (androidx.media3.common.util.Util.SDK_INT > 23) {
            releasePlayer()
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
        loadingProgressBar = null
    }
}